pkg/apis/flowcontrol/validation/validation.go

```diff
@@ -295,27 +348,49 @@ func ValidateFlowSchemaCondition(condition *flowcontrol.FlowSchemaCondition, fld
 // ValidatePriorityLevelConfiguration validates priority-level-configuration.
 func ValidatePriorityLevelConfiguration(pl *flowcontrol.PriorityLevelConfiguration) field.ErrorList {
 	allErrs := apivalidation.ValidateObjectMeta(&pl.ObjectMeta, false, ValidatePriorityLevelConfigurationName, field.NewPath("metadata"))
-	allErrs = append(allErrs, ValidatePriorityLevelConfigurationSpec(&pl.Spec, pl.Name, field.NewPath("spec"))...)
+	specPath := field.NewPath("spec")
+	switch pl.Name {
+	case flowcontrol.PriorityLevelConfigurationNameExempt, flowcontrol.PriorityLevelConfigurationNameCatchAll:
```

Comment:
am in favor of moving this to a dedicated admission-controller

Reply:
We decided to not use an admission controller.
Ensuring the existence and correctness of the mandatory objects is primarily the responsibility of a controller launched from a post-start-hook.  This validation check improves the user experience by forbidding an operation that said controller will silently overwrite.

Reply:
Nothing stops us from invoking `ValidatePriorityLevelConfigurationSpec` in all cases here.  We just have to think about what to do when the result is `false`.  That is probably easier to answer if it is done at startup.  How do I write an `init()` that can fail startup?